src/to_wkt/geo_trait_impl.rs

```diff
@@ -0,0 +1,488 @@
+use core::fmt;
+use std::fmt::Write;
+
+use geo_traits::{
+    CoordTrait, GeometryCollectionTrait, GeometryTrait, LineStringTrait, LineTrait,
+    MultiLineStringTrait, MultiPointTrait, MultiPolygonTrait, PointTrait, PolygonTrait, RectTrait,
+    TriangleTrait,
+};
+
+use crate::error::Error;
+use crate::types::Coord;
+use crate::WktNum;
+
+/// The physical size of the coordinate dimension
+///
+/// This is used so that we don't have to call `.dim()` on **every** coordinate. We infer it once
+/// from the `geo_traits::Dimensions` and then pass it to each coordinate.
+#[derive(Clone, Copy)]
+enum PhysicalCoordinateDimension {
+    Two,
+    Three,
+    Four,
+}
+
+impl From<geo_traits::Dimensions> for PhysicalCoordinateDimension {
+    fn from(value: geo_traits::Dimensions) -> Self {
+        match value.size() {
+            2 => Self::Two,
+            3 => Self::Three,
+            4 => Self::Four,
+            size => panic!("Unexpected dimension for coordinate: {}", size),
+        }
+    }
+}
+
+pub fn write_point<T: WktNum + fmt::Display, G: PointTrait<T = T>, W: Write>(
+    g: &G,
+    f: &mut W,
```

Comment:
```suggestion
    f: &mut impl Write
```
It's a little more readable to use `impl Trait` for these kinds of params.

e.g. 
- https://github.com/rust-lang/rust/blob/dff3e7ccd4a18958c938136c4ccdc853fcc86194/src/tools/rust-analyzer/crates/proc-macro-api/src/json.rs#L29
- https://github.com/rust-lang/rust/blob/dff3e7ccd4a18958c938136c4ccdc853fcc86194/compiler/rustc_pattern_analysis/src/rustc/print.rs#L47

style nit #2: I think conventionally the `impl Write` is the first param for free functions, so all together:

```
pub fn write_point<T: WktNum + fmt::Display, G: PointTrait<T = T>>(f: &mut impl Write, g: &G)

// e.g.
write_point(&mut some_writable, &some_point)
```

(similar for other methods in this file)

Reply:
> It's a little more readable to use `impl Trait` for these kinds of params.

Happy to switch. Can you clarify what you mean by "these kinds"? Do you just mean "simple bounds without any parameters"? Like you switched the `impl Write` but not the `impl PointTrait`

Reply:
I think anything that **can** be written as `impl Trait` probably should be. Maybe there's an exception, but I can't think of one.